Added support for multiple Applications Mapping files (#27)
IBM/dbb-git-migration-modeler: docs/AdvancedInformation.md

~~~~YAML
datasets:
  - APPS.COBOL
  - APPS.COPY
  - APPS.BMS
applications:
  - application: "RetirementCalculator"
    description: "RetirementCalculator"
    owner: "MDALBIN"
    namingConventions:
      - EBU.....
      - LINPUT..
  - application: "GenApp"
    description: "GenApp"
    owner: "DBEHM"
    namingConventions:
      - LGA.....
      - LGD.....
      - LGI.....
      - LGT.....
      - LGU.....
      - LGS.....
      - LGW.....
      - OLD.....
      - FLE.....
  - application: "CBSA"
    description: "CBSA"
    owner: "MDALBIN"
    namingConventions:
      - ABN.....
      - ACC.....
      - BAN.....
      - BNK.....
      - CON.....
      - CRD.....
      - CRE.....
      - CUS.....
      - DBC.....
      - DEL.....
      - DPA.....
      - GET.....
      - INQ.....
      - PRO.....
      - UPD.....
      - XFR.....
      - ZUN.....
~~~~

The result of this command is a set of Application Descriptor files and DBB Migration mapping files for each discovered application.
If a member doesn't match any naming convention, it is assigned to a special application called *UNASSIGNED*.

> ## Working with the special *UNASSIGNED* application

Reviewer: ```suggestion
## Working with source code that is known to be shared
```

> A good strategy could be to store all the shared Include Files in this *UNASSIGNED* application.

Reviewer: ```suggestion
For files that are already known as shared between applications, you can define an application mapping configuration to define their dedicated context. 
```

> This can be done in several ways: as mentioned earlier, all artifacts for which no naming convention is matching will be assigned to this special application.

Reviewer: ```suggestion

```

> Otherwise, if a library is known to contain only shared Include Files, a specific `Applications Mapping` file could be used, as follows:

Reviewer: ```suggestion
If one library already contains these known shared include files, configure a specific `Applications Mapping` file alike the below sample:
```

~~~~YAML
datasets:
  - SHARED.COPY
applications:

>   - application: "UNASSIGNED"

Reviewer: ```suggestion
  - application: "SHARED"
```

1. The [Applications Mapping file](./samples/applicationsMapping.yaml) (YAML format) contains the list of existing applications with their naming convention patterns used for filtering members. It can be created manually or can be filled with information coming from external databases or provided by a report from an SCM solution. Instead of patterns for naming conventions, the file also accepts fully qualified member names that can be extracted from an existing data source or report provided by your legacy tool.  
If no naming convention is applied for a given application, or if all the members of a given dataset belong to the same application, a naming convention whose value is `........` should be defined.
Members in the input PDSs libraries that do not match any convention will be associated to the *UNASSIGNED* application. This is often applicable for include files that do not have an owner assigned.

> Multiple Applications Mapping files can be specified, that define one or multiple application configurations. The DBB Git Migration Modeler will import all the Applications Mapping files first, before processing the mappings. This configuration helps for more granular configurations and advanced scenarios, for instance when the input datasets contain members from only one application or when multiple applications ahve artifacts mixed in the same goup of datasets.

Reviewer: ```suggestion
Multiple Applications Mapping files can be specified, that define one or multiple application configurations. The DBB Git Migration Modeler will import all the Applications Mapping files first, before processing the mappings. This configuration helps for more granular configurations and advanced scenarios, for instance when the input datasets contain members from only one application or when multiple applications have artifacts mixed in the same group of datasets.
```

To configure this case, a specific `Applications Mapping` file for the application should be provided, in the Applications Mappings folder specified by the `DBB_MODELER_APPCONFIG_DIR` parameter. An universal naming convention filter like in the below sample should be used, because all files from the input datasets are mapped to the defined application.

The following is an example of such an `Applications Mapping` YAML file (named *applicationsMapping-CATMAN.yaml*)
~~~~YAML
datasets:
  - CATMAN.COBOL
  - CATMAN.COPY
  - CATMAN.BMS
applications:
  - application: "Catalog Manager"
    description: "Catalog Manager"
    owner: "MDALBIN"
    namingConventions:
      - ........
~~~~

When running the Migration-Modeler-Start.sh script with this Applications Mapping file, all the artifacts found in the input datasets (CATMAN.COBOL, CATMAN.COPY and CATMAN.BMS) will be assigned to the Catalog Manager application. The result of this command is an Application Descriptor file that documents all the artifacts contained in the given datasets, and a DBB Migration mapping file to manages all the members found.

> ### A group of datasets contains artifacts belonging to multiple applications

Reviewer: Let's make this one the first scenario. I believe, it is the most common one.
